pkg/core/proxy/proxy.go

```diff
@@ -333,7 +337,7 @@ func (p *Proxy) handleConnection(ctx context.Context, srcConn net.Conn) error {
 
 		//mock the outgoing message
 		err := p.Integrations["mysql"].MockOutgoing(parserCtx, srcConn, &integrations.ConditionalDstCfg{Addr: dstAddr}, m.(*MockManager), rule.OutgoingOptions)
-		if err != nil {
+		if err != nil && !p.IsUserAppTerminateInitiated() {
```

Comment:
Instead of here . Cancel the context [here](https://github.com/keploy/keploy/blob/e35909158d3728b25d58304be55ba1ff7c7e2972/pkg/core/proxy/proxy.go#L235) so that errors will automatically taken care.

Reply:
should I cancel "clientConnCtx" ?